components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/model/OpenAPIRequest.java

```diff
@@ -146,4 +168,13 @@ public Method getMethod() {
         return SchemaValidationUtils.getFromMapOrEmptyList(this.headers.asMap(), s);
     }
 
+    protected void validatePath(OpenAPI openAPI) {
+
+        Paths paths = openAPI.getPaths();
+        if (path.equals("/") && !paths.containsKey(path)) {
+            if (paths.containsKey("/*")) {
+                path = "/*";
+            }
+        }
+    }
```

Comment:
do we have tests for this? we can easily write unit tests, right?

Reply:
yes, will write unit test